pkg/annotations/ingress_annotations.go

```diff
@@ -43,6 +43,10 @@ const (
 	// ApplicationGatewayIngressClass defines the value of the `IngressClassKey` annotation that will tell the ingress controller
 	// whether it should act on this ingress resource or not.
 	ApplicationGatewayIngressClass = "azure/application-gateway"
+
+	// ApplicationGatewayIstioIngressClass defines the value of the `IngressClassKey` annotation that will tell the ingress controller whether
+	// this ingress resource is from Istio or not.
+	ApplicationGatewayIstioIngressClass = "istio-ingress"
```

Comment:
Here is what a simple definition of an Istio Gateway might look like:
```yaml
kubectl apply -f - <<EOF
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: httpbin-gateway
spec:
  selector:
    istio: ingressgateway # use Istio default gateway implementation
  servers:
  - port:
      number: 80
      name: http
      protocol: HTTP
    hosts:
    - "httpbin.example.com"
```

With the proposed annotation here, we'd see the following added to the YAML above:
```yaml
metadata:
    annotations:
        kubernetes.io/ingress.class: istio-ingress
```

I don't think that's quite right.

I think that we can re-use pieces of our existing `ApplicationGatewayIngressClass` annotation but put it in the Istio namespace (to be semantically correct... not that that string really matters).

How about:
```yaml
kubectl apply -f - <<EOF
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: httpbin-gateway
  annotations:
    # option A - create a new tag - namespaced within istio.io
    istio.io/ingress.class: azure/application-gateway
    # option B - reuse the existing tag we have
    kubernetes.io/ingress.class: azure/application-gateway
spec:
  selector:
    istio: ingressgateway # use Istio default gateway implementation
  servers:
  - port:
      number: 80
      name: http
      protocol: HTTP
    hosts:
    - "httpbin.example.com"
```